src/chains/filecoin/filecoin/src/things/version.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  SerializableObject,
+  DeserializedObject,
+  SerializedObject,
+  Definitions
+} from "./serializable-object";
+import { $INLINE_JSON } from "ts-transformer-inline-file";
+
+const { version: GanacheFilecoinVersion } = $INLINE_JSON("../../package.json");
+
+function createBinarySemverVersion(version: string): number {
+  const versionParts = version.split(".");
+
+  const majorVersion =
+    versionParts.length > 0 ? parseInt(versionParts[0], 10) : 0;
+  const minorVersion =
+    versionParts.length > 1 ? parseInt(versionParts[1], 10) : 0;
+  const patchVersion =
+    versionParts.length > 2 ? parseInt(versionParts[2], 10) : 0;
+
+  const binaryVersion =
+    (majorVersion << 16) | (minorVersion << 8) | patchVersion;
+
+  return binaryVersion;
+}
+
+interface VersionConfig {
+  properties: {
+    version: {
+      type: string;
+      serializedType: string;
+      serializedName: "Version";
+    };
+    apiVersion: {
+      type: number;
+      serializedType: number;
+      serializedName: "APIVersion";
+    };
+    blockDelay: {
+      type: bigint;
+      serializedType: string;
+      serializedName: "BlockDelay";
+    };
+  };
+}
+
+class Version
+  extends SerializableObject<VersionConfig>
+  implements DeserializedObject<VersionConfig> {
+  get config(): Definitions<VersionConfig> {
+    return {
+      version: {
+        serializedName: "Version",
+        defaultValue: `@ganache/filecoin v${GanacheFilecoinVersion}`
+      },
+      apiVersion: {
+        serializedName: "APIVersion",
+        defaultValue: createBinarySemverVersion("0.0.1") // TODO: not sure what to put here
```

Comment:
as far as the "0.0.1" part goes, maybe just use the `GanacheFilecoinVersion` semver version?

Reply:
Since writing this, i now believe it should be `1.4.0`, which is found at the top of https://pkg.go.dev/github.com/filecoin-project/lotus/api